Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractCompactRandomCutTree.java

```diff
@@ -77,7 +80,12 @@ public AbstractCompactRandomCutTree(
         }
 
         if (builder.root == NULL) {
-            this.nodeStore = new NodeStore(maxSize - 1);
+            if (builder.float32Precision && maxSize < SmallNodeStore.MAX_SMALLNODESTORE_CAPACITY
```

Comment:
Can we extract a method to check if small node store is needed? Saw this logic used in a couple of places.

Reply:
Again, great callout. Changing.